src/main/webapp/location/keyword/container.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react'
+import { geometry } from 'geospatialdraw'
+import { useLazyQuery } from '@apollo/react-hooks'
+import gql from 'graphql-tag'
+import {
+  Suggestion,
+  KeywordGeoProperties,
+  ContainerProps as Props,
+} from './props'
+import Keyword from './presentation'
+
+const MINIMUM_INPUT_LENGTH = 2
+
+const SuggestionsQuery = gql`
+  query SuggestionsQuery($q: String!) {
+    suggestions(q: $q) {
+      id
+      name
+    }
+  }
+`
+
+const FeatureQuery = gql`
+  query FeatureQuery($id: String!) {
+    geofeature(id: $id) {
+      type
+      geometry
+      properties
+      id
+    }
+  }
+`
+
+type suggestionResult = {
+  suggestions: Suggestion[]
+}
+
+type geofeatureResult = {
+  geofeature: geometry.GeometryJSON
+}
+
+const Container: React.SFC<Props> = ({
+  value,
+  onChange,
+  minimumInputLength = MINIMUM_INPUT_LENGTH,
+  ...rest
+}) => {
+  const {
+    keyword = '',
+    keywordId = '',
+  } = value.properties as KeywordGeoProperties
+  const [selectedId, setSelectedId] = React.useState('')
```

Comment:
I think you can move this state higher up in the component tree

Reply:
both input and selectedId

Reply:
There is no higher up component. The keyword container is just a GeoJSON in GeoJSON out interface. Internal details are not exposed this. This matches the interface provided by the other location input fields allowing for polymorphism.